packages/koa-shopify-auth/src/auth/create-request-storage-access.ts

```diff
@@ -0,0 +1,82 @@
+import {Context} from 'koa';
+
+import {OAuthStartOptions} from '../types';
+
+import css from './client/polaris-css';
+import itpHelper from './client/itp-helper';
+import requestStorageAccess from './client/request-storage-access';
+import storageAccessHelper from './client/storage-access-helper';
+import Error from './errors';
+
+const HEADING = 'This app needs access to your browser data';
+const BODY =
+  'Your browser is blocking this app from accessing your data. To continue using this app, click Continue, then click Allow if the browser prompts you.';
+const ACTION = 'Continue';
+
+export default function createRequestStorageAccess({
+  apiKey,
+}: OAuthStartOptions) {
+  return function requestStorage(ctx: Context) {
+    const {query} = ctx;
+    const {shop} = query;
+
+    if (shop == null) {
+      ctx.throw(400, Error.ShopParamMissing);
+      return;
+    }
+
+    ctx.body = `
```

Comment:
Eventually we might want to introduce tagged template literals or something so we can make all of these a bit nicer, but this is fine for now.